File: 5_4/p067.py
h, w = map(int, input().split())
R = [0]*h
L = [0]*w
A = [list(map(int, input().split())) for _ in range(h)]
for i in range(h):
    sum_num = 0
    for j in range(w):
        sum_num += A[i][j]
    R[i] = sum_num

for j in range(w):
    sum_num = 0
    for i in range(h):
        sum_num += A[i][j]
    L[j] = sum_num

for i in range(h):
    ans = [0]*w
    for j in range(w):
        ans[j] = str(R[i] + L[j] - A[i][j])
    print(' '.join(ans))

File: 5_4/p067.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

var scan = bufio.NewScanner(os.Stdin)

const (
	initBufSize = 10000
	maxBufSize = 10000000
)

func init() {
	buf := make([]byte, initBufSize)
	scan.Buffer(buf, maxBufSize)
	scan.Split(bufio.ScanWords)
}

func main() {
	h := NumStdin()
	w := NumStdin()

	area_y := make([]int, h)
	area_x := make([]int, w)
	in_lst := make([][]int, h)
	for i := 0; i < h; i++ {
		row := make([]int, w)
		row_sum := 0
		for j := 0; j < w; j++ {
			num := NumStdin()
			row[j] = num
			row_sum += num
			area_x[j] += num
		}
		in_lst[i] = row
		area_y[i] = row_sum
	}

	for i := 0; i < h; i++ {
		row := make([]string, w)
		for j := 0; j < w; j++ {
			row[j] = strconv.Itoa(area_y[i] + area_x[j] - in_lst[i][j])
		}
		fmt.Println(strings.Join(row, " "))
	}

}

func NumStdin() int {
	scan.Scan()
	num, err := strconv.Atoi(scan.Text())
	if err != nil {
		panic(err)
	}
	return num
}
